tests/test-gpg-verify-result.c

```diff
@@ -373,14 +417,91 @@ test_expired_signature (TestFixture *fixture,
                  &key_missing,
                  &key_exp_timestamp);
 
-  g_assert_true (valid);
+  g_assert_false (valid);
```

Comment:
It's because of the populated `trustdb.gpg` that's in the repo right now which I emptied out in this commit. When you generate a private key, GPG will mark it as ultimately trusted. An signature by an ultimately trusted key is still valid even if the key is expired. After I emptied the trustdb, the keys are not marked with any trust and the signature from an expired key becomes a failure. In essence, the trustdb was causing an incorrect answer for this test.

Reply:
If you want I can break that out to a separate PR. That and a few other commits are just test cleanup.

Reply:
I think the behaviour change makes sense, though might be worth a comment here about it so we can link to it in case anyone else somehow was hitting this.

Reply:
It's not actually a behavior change, it's just how gnupg handles trust. The change here is just trying to avoid gnupg's default behavior. This would only affect people that have the ostree verification keys in their GPG homedir's trustdb. I think that's very unlikely but still possible.

Taking a look at the GPGME manual, it seems that since 0.11.2 you can [set the trust model](https://www.gnupg.org/documentation/manuals/gpgme/Context-Flags.html) on the context. The [gnupg manual](https://www.gnupg.org/documentation/manuals/gnupg/GPG-Configuration-Options.html#index-trust_002dmodel) has a description of the models. I believe a model of `always` would match what ostree is trying to do. Essentially, all keys in the trusted keyring are trusted as valid unless they're expired or revoked.

Reply:
Right, I got mixed up. It's just a consequence of the change to the trustdb test fixture.